includes/preflight.inc

```diff
@@ -386,7 +389,8 @@ function _drush_find_commandfiles_drush() {
     }
   }
 
-  if (!drush_get_option('local')) {
+  // In site-local Drush mode, we ignore global commandfiles.
+  if (!SITE_LOCAL_DRUSH && !drush_get_option('site-local')) {
```

Comment:
This looks OK but its going to annoy folks whose team uses site-local drush but they still want to use global commands. I guess those folks could use configuration to add another search path that points to global commands? if so, this PR looks good.